src/pykiso/lib/auxiliaries/udsaux/uds_auxiliary.py

```diff
@@ -0,0 +1,382 @@
+##########################################################################
+# Copyright (c) 2010-2022 Robert Bosch GmbH
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0
+##########################################################################
+
+"""
+uds_auxiliary
+*************
+
+:module: uds_auxiliary
+
+:synopsis: Auxiliary used to handle Unified Diagnostic Service protocol
+
+.. currentmodule:: uds_auxiliary
+
+"""
+import configparser
+import logging
+import threading
+from pathlib import Path
+from typing import List, Optional, Union
+
+import can
+
+from pykiso.connector import CChannel
+from pykiso.interfaces.thread_auxiliary import AuxiliaryInterface
+
+from . import uds_exceptions
+from .uds_request import UDSCommands
+from .uds_response import UdsResponse
+from .uds_utils import get_uds_service
+
+log = logging.getLogger(__name__)
+
+Uds = None
+createUdsConnection = None
+IsoServices = None
+
+
+class UdsAuxiliary(AuxiliaryInterface):
+    """Auxiliary used to handle UDS messages"""
+
+    NEGATIVE_RESPONSE = 0x7F
+    POSITIVE_RESPONSE_OFFSET = 0x40
+    errors = uds_exceptions
+
+    def __new__(cls, *args, **kwargs):
+        """Handle class creation.
+
+        Import python-uds at instanciation to avoid warnings due to
+        missing drivers.
+        """
+
+        try:
+            global Uds, createUdsConnection, IsoServices
+            from uds import IsoServices, Uds, createUdsConnection
+
+            Uds = Uds
+            createUdsConnection = createUdsConnection
+            IsoServices = IsoServices
+        except ImportError:
+            log.exception("Could not import python-uds")
+        return super(UdsAuxiliary, cls).__new__(cls)
+
+    def __init__(
+        self,
+        com: CChannel,
+        odx_file_path: str,
```

Comment:
Can you please make this param optional? It doesn't really make sense to keep it mandatory just to explicitely set it to None